internal/supervisor/group_test.go

```diff
@@ -0,0 +1,106 @@
+package supervisor_test
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-cloud/internal/store"
+	"github.com/mattermost/mattermost-cloud/internal/supervisor"
+	"github.com/mattermost/mattermost-cloud/internal/testlib"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/stretchr/testify/require"
+)
+
+type mockGroupStore struct {
+	Group                     *model.Group
+	UnlockedGroupsPendingWork []*model.Group
+	GroupRollingMetadata      *store.GroupRollingMetadata
+
+	Installation *model.Installation
+
+	UnlockChan              chan interface{}
+	UpdateInstallationCalls int
+}
+
+func (s *mockGroupStore) GetUnlockedGroupsPendingWork() ([]*model.Group, error) {
+	return s.UnlockedGroupsPendingWork, nil
+}
+
+func (s *mockGroupStore) GetGroupRollingMetadata(groupID string) (*store.GroupRollingMetadata, error) {
+	return s.GroupRollingMetadata, nil
+}
+
+func (s *mockGroupStore) LockGroup(groupID, lockerID string) (bool, error) {
+	return true, nil
+}
+
+func (s *mockGroupStore) UnlockGroup(groupID, lockerID string, force bool) (bool, error) {
+	if s.UnlockChan != nil {
+		close(s.UnlockChan)
+	}
+	return true, nil
+}
+
+func (s *mockGroupStore) GetInstallation(installationID string, includeGroupConfig, includeGroupConfigOverrides bool) (*model.Installation, error) {
+	return s.Installation, nil
+}
+
+func (s *mockGroupStore) UpdateInstallation(installation *model.Installation) error {
+	s.UpdateInstallationCalls++
+	return nil
+}
+
+func (s *mockGroupStore) UpdateInstallationState(id, state string) error {
+	s.UpdateInstallationCalls++
+	return nil
+}
+
+func (s *mockGroupStore) LockInstallation(installationID, lockerID string) (bool, error) {
+	return true, nil
+}
+
+func (s *mockGroupStore) UnlockInstallation(installationID, lockerID string, force bool) (bool, error) {
+	if s.UnlockChan != nil {
+		close(s.UnlockChan)
+	}
+	return true, nil
+}
+
+func TestGroupSupervisorDo(t *testing.T) {
```

Comment:
Do we have a test case that I'm overlooking that exercises the maxRolling feature/setting?

Reply:
Nothing directly for that yet. I have the stub for the test here and will update it on the final PR when all the group merge stuff is turned on.